index.js

```diff
@@ -1,11 +1,277 @@
+          }
+        ],
+        def: false
+      },
+      startTime: {
+        label: 'Start time',
+        type: 'time',
+        def: '09:00 AM',
+        required: true,
+        if: {
+          allDay: false
+        }
+      },
+      endTime: {
+        label: 'End time',
+        type: 'time',
+        def: '05:30 PM',
+        required: true,
+        if: {
+          allDay: false
+        }
+      },
+      dateType: {
+        label: 'What type of event is this?',
+        help:
+          'Select if the event is on a single day, consecutive days, or repeats.',
+        type: 'select',
+        choices: [
+          {
+            label: 'Single Day',
+            value: 'single'
+          },
+          {
+            label: 'Consecutive Days',
+            value: 'consecutive'
+          },
+          {
+            label: 'Recurring',
+            value: 'repeat'
+          }
+        ],
+        def: 'single'
+      },
+      endDate: {
+        label: 'End date',
+        type: 'date',
+        if: {
+          dateType: 'consecutive'
+        }
+      },
+      repeatInterval: {
+        label: 'How often does the event repeat?',
+        type: 'select',
+        choices: [
+          {
+            label: 'Every week',
+            value: 'weeks'
+          },
+          {
+            label: 'Every month',
+            value: 'months'
+          }
+        ],
+        if: {
+          dateType: 'repeat'
+        }
+      },
+      repeatCount: {
+        label: 'How many times does it repeat?',
+        type: 'integer',
+        def: 1,
+        if: {
+          dateType: 'repeat'
+        }
+      },
+      description: {
+        type: 'string',
+        label: 'Description',
+        textarea: true,
+        required: true
+      },
+      image: {
+        label: 'Headline photo',
+        type: 'area',
+        options: {
+          max: 1,
+          widgets: {
+            '@apostrophecms/image': {}
+          }
+        },
+        required: false
+      }
+    },
+    group: {
+      basics: {
+        label: 'Basics',
+        fields: [
+          'title',
+          'slug',
+          'description',
+          'image',
+          'startDate',
+          'allDay',
+          'startTime',
+          'endTime'
+        ]
+      },
+      advanced: {
+        label: 'Advanced',
+        fields: ['dateType', 'endDate', 'repeatInterval', 'repeatCount']
+      },
+      meta: {
+        label: 'Meta',
+        fields: ['tags', 'published']
+      }
+    }
+  },
+  handlers(self, options) {
+    return {
+      beforeSave: {
+        async denormalizeDateTimes(req, piece, options) {
+          self.denormalizeDatesAndTimes(piece);
+        }
+      },
+      beforeInsert: {
+        setGroupId(req, piece, options) {
+          // Set groupId on parent if this is a repeating item
+          if (
+            piece.dateType === 'repeat' &&
+            !piece.groupId &&
+            !self._workflowPropagating
```

Comment:
This flag does not exist in 3.x. I suggest checking to make sure piece.aposMode is draft instead.

Reply:
Fixed